test_conformance/extensions/cl_khr_command_buffer/negative_command_buffer_svm_mem.cpp

```diff
@@ -0,0 +1,413 @@
+            "clCommandSVMMemcpyKHR should return CL_INVALID_COMMAND_QUEUE",
+            TEST_FAIL);
+
+        error = clCommandSVMMemFillKHR(command_buffer, queue, svm_in_mem(),
+                                       &pattern_1, sizeof(cl_char), data_size(),
+                                       0, nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(
+            error, CL_INVALID_COMMAND_QUEUE,
+            "clCommandSVMMemFillKHR should return CL_INVALID_COMMAND_QUEUE",
+            TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+
+    bool Skip() override
+    {
+        return BasicSVMCommandBufferTest::Skip()
+            || is_extension_available(device,
+                                      "cl_khr_command_buffer_multi_device");
+    }
+
+    const cl_char pattern_1 = 0x14;
+};
+
+// CL_INVALID_CONTEXT if the context associated with command_queue,
+// command_buffer, and kernel are not the same.
+struct CommandBufferCommandSVMDifferentContext
+    : public BasicSVMCommandBufferTest
+{
+    using BasicSVMCommandBufferTest::BasicSVMCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_int error = clCommandSVMMemcpyKHR(
+            command_buffer, nullptr, svm_out_mem(), svm_in_mem(), data_size(),
+            0, nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(
+            error, CL_INVALID_CONTEXT,
+            "clCommandSVMMemcpyKHR should return CL_INVALID_CONTEXT",
+            TEST_FAIL);
+
+        error = clCommandSVMMemFillKHR(command_buffer, nullptr, svm_in_mem(),
+                                       &pattern_1, sizeof(cl_char), data_size(),
+                                       0, nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(
+            error, CL_INVALID_CONTEXT,
+            "clCommandSVMMemFillKHR should return CL_INVALID_CONTEXT",
+            TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+
+    bool Skip() override
+    {
+        return BasicSVMCommandBufferTest::Skip()
+            || is_extension_available(device,
+                                      "cl_khr_command_buffer_multi_device");
+    }
+
+    cl_int SetUp(int elements) override
+    {
+        cl_int error = BasicCommandBufferTest::SetUp(elements);
+        test_error(error, "BasicCommandBufferTest::SetUp failed");
+
+        error = CreateKernelWithDifferentContext();
+        test_error(error, "Failed to create kernel");
+
+        size_t size = sizeof(cl_int) * num_elements * buffer_size_multiplier;
+        svm_in_mem = clSVMWrapper(new_context, size);
+        if (svm_in_mem() == nullptr)
+        {
+            test_error(CL_OUT_OF_RESOURCES, "Unable to allocate SVM memory");
+        }
+        svm_out_mem = clSVMWrapper(new_context, size);
+        if (svm_out_mem() == nullptr)
+        {
+            test_error(CL_OUT_OF_RESOURCES, "Unable to allocate SVM memory");
+        }
+
+
+        return CL_SUCCESS;
+    }
+
+    cl_int CreateKernelWithDifferentContext()
```

Comment:
There's no kernel parameter to these SVM functions, the existence of this error wording is a mistake that should be fixed by https://github.com/KhronosGroup/OpenCL-Docs/pull/1149 So we can remove this kernel creation code.

Reply:
@EwanC I see that "kernel" word is a mistake as there is no parameter for kernel. Then when INVALID_CONTEXT should be returned? Should I pass not null command_queue created with different context and command_buffer created with different command_queue (with different context) ? What do you think about it?

Reply:
> Should I pass not null command_queue created with different context and command_buffer created with different command_queue (with different context) ? What do you think about it?

I would have suggested that, but Ben raised I point I hadn't considered in https://github.com/KhronosGroup/OpenCL-CTS/pull/1956#discussion_r1606114591 so it's fine to leave it out for now.